Code/Framework/AzToolsFramework/AzToolsFramework/Viewport/ViewBookmarkLoaderInterface.h

```diff
@@ -0,0 +1,87 @@
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#pragma once
+
+#include <AzCore/Interface/Interface.h>
+#include <AzCore/Serialization/SerializeContext.h>
+
+namespace AzToolsFramework
+{
+    //! @class ViewBookmark
+    //! @brief struct that store viewport camera properties that can be serialized and loaded
+    struct ViewBookmark
+    {
+        AZ_CLASS_ALLOCATOR(ViewBookmark, AZ::SystemAllocator, 0);
+        AZ_TYPE_INFO(ViewBookmark, "{522A38D9-6FFF-4B96-BECF-B4D0F7ABCD25}");
+
+        ViewBookmark() = default;
+
+        static void Reflect(AZ::ReflectContext* context)
+        {
+            if (auto* serializeContext = azrtti_cast<AZ::SerializeContext*>(context))
+            {
+                serializeContext->Class<ViewBookmark>()
+                    ->Version(0)
+                    ->Field("Position", &ViewBookmark::m_position)
+                    ->Field("Rotation", &ViewBookmark::m_rotation);
+
+                if (AZ::EditContext* editContext = serializeContext->GetEditContext())
+                {
+                    editContext->Class<ViewBookmark>("ViewBookmark Data", "")
+                        ->ClassElement(AZ::Edit::ClassElements::EditorData, "ViewBookmark")
+                        ->Attribute(AZ::Edit::Attributes::AutoExpand, true)
+                        ->DataElement(AZ::Edit::UIHandlers::Vector3, &ViewBookmark::m_position, "Position", "")
+                        ->DataElement(AZ::Edit::UIHandlers::Vector3, &ViewBookmark::m_rotation, "Rotation", "");
+                }
+            }
+        }
+
+        bool operator==(const ViewBookmark& other) const
+        {
+            return m_position == other.m_position && m_rotation == other.m_rotation;
+        }
+
+        bool operator!=(const ViewBookmark& other) const
+        {
+            return m_position != other.m_position || m_rotation != other.m_rotation;
+        }
+
+        AZ::Vector3 m_position = AZ::Vector3::CreateZero();
+        AZ::Vector3 m_rotation = AZ::Vector3::CreateZero();
+    };
+
+    //! @class ViewBookmarkIntereface
+    //! @brief Interface for saving/loading View Bookmarks.
+    class ViewBookmarkLoaderInterface
+    {
+    public:
+        AZ_RTTI(ViewBookmarkLoaderInterface, "{71E7E178-4107-4975-A6E6-1C4B005C981A}")
+
+        //! Choose storage mode for ViewBookmarks
+        //! Shared are stored in the prefab.
+        //! Local are stored in the Settings Registry.
+        enum class StorageMode : int
+        {
+            Shared = 0,
+            Local = 1,
+            Invalid = -1
+        };
+
+        virtual bool SaveBookmark(ViewBookmark bookmark, const StorageMode mode) = 0;
+        virtual bool ModifyBookmarkAtIndex(ViewBookmark bookmark, int index, const StorageMode mode) = 0;
+        virtual bool SaveLastKnownLocation(ViewBookmark bookmark) = 0;
```

Comment:
These functions could pass ViewBookmark as const reference to avoid making copied each time they are called.